File: string_art_gui/lib/models/draw_backend.dart
import 'dart:ui';

import 'package:string_art_gui/models/basic_types.dart';

abstract mixin class DrawBackend {
  void drawSegment(Segment segment, double stroke, Color color);

  void drawCircle(Offset offset, double radius, Color color);

  void drawCircunference(
    Offset offset,
    double radius,
    double stroke,
    Color color,
  );
}

File: string_art/src/draw_backend.rs
use string_art_color::Color32;
use string_art_geometry::{Circle, Segment};

pub trait DrawBackend {
    fn draw_circle(&mut self, circle: Circle<f32>, color: impl Color32);

    fn draw_circunference(&mut self, circle: Circle<f32>, stroke: f32, color: impl Color32);

    fn draw_segment(&mut self, segment: Segment<f32>, stroke: f32, color: impl Color32);
}
